File: javascript/121_protein-translation/protein-translation.js
const CODON_MAPA = {
  'AUG': 'Methionine',
  'UUU': 'Phenylalanine',
  'UUC': 'Phenylalanine',
  'UUA': 'Leucine',
  'UUG': 'Leucine',
  'UCU': 'Serine',
  'UCC': 'Serine',
  'UCA': 'Serine',
  'UCG': 'Serine',
  'UAU': 'Tyrosine',
  'UAC': 'Tyrosine',
  'UGU': 'Cysteine',
  'UGC': 'Cysteine',
  'UGG': 'Tryptophan'
}
const STOP = ['UAA', 'UAG', 'UGA'];

export const translate = (input = '') => {
  const output = [];
  for (let i = 0; i < input.length; i += 3) {
    const next = input.slice(i, i + 3);
    if(STOP.includes(next)) break;
    if(!(next in CODON_MAPA)) throw new Error('Invalid codon'); 
    output.push(CODON_MAPA[next]);
  }
  return output;
};

File: python/ex_057_protein-translation/protein_translation.py
codon_protein_map = {key:value for key_list, value in [row.split(':') 
					for row in 'AUG:Methionine;UUU,UUC:Phenylalanine;UUA,UUG:Leucine;UCU,UCC,UCA,UCG:Serine;UAU,UAC:Tyrosine;UGU,UGC:Cysteine;UGG:Tryptophan;UAA,UAG,UGA:STOP'.split(';')] 
					for key in key_list.split(',')}
def proteins(strand):
    output = []
    for i in range(0,len(strand),3):
        protein = codon_protein_map.get(strand[i:i+3],'STOP')
        if protein == 'STOP': break
        output.append(protein)
    return output